script/20240131-maptoken-pixel/20240131-maptoken-pixel-mainchain.s.sol

```diff
@@ -0,0 +1,142 @@
+    uint256[][4] memory thresholds;
+
+    uint256 expiredTime = block.timestamp + 10 days;
+    address[] memory targets = new address[](4);
+    uint256[] memory values = new uint256[](4);
+    bytes[] memory calldatas = new bytes[](4);
+    uint256[] memory gasAmounts = new uint256[](4);
+
+    // ================ PIXEL ERC-20 ======================
+
+    mainchainTokens[0] = _pixelMainchainToken;
+    roninTokens[0] = _pixelRoninToken;
+    standards[0] = Token.Standard.ERC20;
+    // highTierThreshold
+    thresholds[0] = new uint256[](1);
+    thresholds[0][0] = _highTierThreshold;
+    // lockedThreshold
+    thresholds[1] = new uint256[](1);
+    thresholds[1][0] = _lockedThreshold;
+    // unlockFeePercentages
+    thresholds[2] = new uint256[](1);
+    thresholds[2][0] = _unlockFeePercentages;
+    // dailyWithdrawalLimit
+    thresholds[3] = new uint256[](1);
+    thresholds[3][0] = _dailyWithdrawalLimit;
+
+    // function mapTokensAndThresholds(
+    //   address[] calldata _mainchainTokens,
+    //   address[] calldata _roninTokens,
+    //   Token.Standard[] calldata _standards,
+    //   uint256[][4] calldata _thresholds
+    // )
+
+    bytes memory innerData = abi.encodeCall(IMainchainGatewayV3.mapTokensAndThresholds, (
+      mainchainTokens,
+      roninTokens,
+      standards,
+      thresholds
+    ));
+
+    bytes memory proxyData = abi.encodeWithSignature("functionDelegateCall(bytes)", innerData);
+
+    targets[0] = _mainchainGatewayV3;
+    values[0] = 0;
+    calldatas[0] = proxyData;
+    gasAmounts[0] = 1_000_000;
+
+    // ================ FARMLAND ERC-721 ======================
+
+    mainchainTokens[0] = _farmlandMainchainToken;
+    roninTokens[0] = _farmlandRoninToken;
+    standards[0] = Token.Standard.ERC721;
+
+    // function mapTokens(
+    //   address[] calldata _mainchainTokens,
+    //   address[] calldata _roninTokens,
+    //   Token.Standard[] calldata _standards
+    // ) external;
+
+    innerData = abi.encodeCall(IMainchainGatewayV3.mapTokens, (
+      mainchainTokens,
+      roninTokens,
+      standards
+    ));
+
+    proxyData = abi.encodeWithSignature("functionDelegateCall(bytes)", innerData);
+
+    targets[1] = _mainchainGatewayV3;
+    values[1] = 0;
+    calldatas[1] = proxyData;
+    gasAmounts[1] = 1_000_000;
+
+    // =============== AXIE CHAT UPDATE ===========
+    targets[2] = _mainchainBridgeManager;
+    values[2] = 0;
+    calldatas[2] = _addAxieChatGovernorAddress();
+    gasAmounts[2] = 1_000_000;
+
+    targets[3] = _mainchainBridgeManager;
+    values[3] = 0;
+    calldatas[3] = _removeAxieChatGovernorAddress();
+    gasAmounts[3] = 1_000_000;
+
+    // ================ VERIFY AND EXECUTE PROPOSAL ===============
+
+    _verifyMainchainProposalGasAmount(targets, values, calldatas, gasAmounts);
+
+    uint256 chainId = _config.getCompanionNetwork(_config.getNetworkByChainId(block.chainid)).chainId();
+
+    vm.broadcast(_governor);
```

Comment:
We can use `vm.broadcast(sender())` to avoid hardcoding _governor

Reply:
Actually I need it for simulation, already check with the current governor list.